plugins/out_s3/s3.c

```diff
@@ -101,6 +115,44 @@ struct flb_aws_header *create_canned_acl_header(char *canned_acl)
     return acl_header;
 };
 
+static struct flb_aws_header *create_headers(struct flb_s3 *ctx, int num_headers)
+{
+    int n = 0;
+    struct flb_aws_header *canned_acl_header = NULL;
+    struct flb_aws_header *headers = NULL;
+    if (num_headers == 0) {
+        return NULL;
+    }
+
+    headers = flb_malloc(sizeof(struct flb_aws_header) * num_headers);
+    if (headers == NULL) {
+        flb_errno();
+        return NULL;
+    }
+
+    if (ctx->content_type != NULL) {
+        memcpy(&headers[n], &content_type_header, sizeof(content_type_header));
+        headers[n].val = ctx->content_type;
+        headers[n].val_len = strlen(ctx->content_type);
+        n++;
+    }
+    if (ctx->compression != NULL) {
+        headers[n] = content_encoding_header;
```

Comment:
you need to use memcpy here too

Reply:
headers is a block of memory that can store some number of flb_aws_headers structs. Its actually just raw memory:

```
flb_malloc(sizeof(struct flb_aws_header) * num_headers);
```

sizeof(struct flb_aws_header) just returns the number of bytes needed for one struct, so there's nothing special about your block of memory

Reply:
its just memory. 

And you need to copy the structure into this memory. In C, the `=` operator can only be used to copy memory for primitive types as far as I know. For structures, you must use memory copying functions.

Reply:
I see. Thanks.